packages/react-composites/src/composites/MeetingComposite/adapter/MeetingAdapter.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/* eslint-disable @typescript-eslint/adjacent-overload-signatures */
+
+import {
+  CallAdapter,
+  CallIdChangedListener,
+  DisplayNameChangedListener,
+  IsMuteChangedListener,
+  IsScreenSharingOnChangedListener,
+  IsSpeakingChangedListener,
+  ParticipantJoinedListener,
+  ParticipantLeftListener
+} from '../../CallComposite';
+import { ChatAdapter, MessageReadListener, MessageReceivedListener, MessageSentListener } from '../../ChatComposite';
+import { MeetingState } from '../state/MeetingState';
+
+export type ConflictingProps = 'getState' | 'onStateChange' | 'offStateChange' | 'on' | 'off';
```

Comment:
Would like to hear more opinions here.

I would argue for an alternative, more verbose, approach: Define two interfaces `ThinCallAdapter` and `ThinChatAdapter` that explicitly redefines all the props we want to pull in, but uses `Pick` to make sure we can't possibly add props that weren't there in the first place. Then, have `MeetingAdapter extends ThinCallAdapter, ThinChatAdapter {...}`.

Reasoning:
* It is easier to understand for the user of `MeetingAdapter`
  * They don't have to go looking for definitions of `CallAdapter` and `ChatAdapter` elsewhere. Everything is listed right here.
  * They don't have to deal with negation logic to figure out what properties exist, e.g. `CallAdapter` has a prop -> It was dropped in the `Omit` -> It was re-added in the `MeetingAdapter`. Switching between additive / subtractive logic is difficult.
* It avoids accidentally expanding the `MeetingComposite` API as the `CallAdapter` and `ChatAdapter` APIs evolve. Conflicting names will get caught by typescript, but that's only one kind of conflict. There might be differently named properties (in the future) that are semantically in conflict. By whitelisting, we must intentionally expand the `MeetingComposite` API each time we want to.

It is more verbose, but we're only redefining the interface. The implementation still gets to share it all.  And the use of `Pick` ensure that we don't accidentally add properties that don't exist in the underlying composites.

wdyt?

Reply:
I like that much more also 👌

Reply:
Updated; broke down the current Chat and Calling Adapter interfaces into smaller extendable interfaces, then used `Pick` on the methods that Meeting wants to implement.
Can we do this also with the `on` methods? I couldn't find a way given how they're defined